### PR TITLE
Support Swift 3 and Xcode 8.

### PR DESCRIPTION
**Pull request by yoichitgy**
Migrated the project to support Swift 3 and Xcode 8.

This pull request consists of 
- https://github.com/yoichitgy/SwinjectPropertyLoader/pull/1 by @gasthana0
- Some cherry-picked commits of #3

### CODE REVIEW

**Comment by yoichitgy** [comment]
Need to fix .travis.yml.

File: Sources/ResolverType+Properties.swift [hunk 1]
```diff
@@ -9,26 +9,26 @@
...
+    fileprivate static var properties: UInt8 = 0
 }
 
 extension ResolverType {
-    private var properties: [String: AnyObject] {
+    fileprivate var properties: [String: AnyObject] {
         get {
-            guard let obj = self as? AnyObject else {
-                fatalError("Property feature is not supported unless self is AnyObject.")
-            }
```
Comment (yoichitgy) [inline Sources/ResolverType+Properties.swift:20]: Need to check this part can be removed completely.

File: Sources/ResolverType+Properties.swift [hunk 2]
```diff
@@ -9,26 +9,26 @@
...
             
-            return objc_getAssociatedObject(obj, &AssociatedKeys.properties) as? [String: AnyObject] ?? [:]
+            return objc_getAssociatedObject(self, &AssociatedKeys.properties) as? [String: AnyObject] ?? [:]
         }
     }
     
-    private func setProperties(newProperties: [String: AnyObject]) {
-        guard let obj = self as? AnyObject else {
-            fatalError("Property feature is not supported unless self is AnyObject.")
-        }
```
Comment (yoichitgy) [inline Sources/ResolverType+Properties.swift:29]: Need to check this part can be removed completely.

File: Sources/ResolverType+Properties.swift (hunk 1 above)
Comment (jakubvano) [inline Sources/ResolverType+Properties.swift]: If this constrain would be required, it is cleaner to express it as
```
extension ResolverType where Self: AnyObject {
    ...
}
```

**Comment by yoichitgy** [comment]
Thanks @jakubvano, I guess I added the `AnyObject` check by my misunderstanding to use an objc associated object. I'll check.

**Comment by yoichitgy** [comment]
Removed the commented out code by be48d02 because the type check was just misunderstanding about AnyObject, NSObject and requirement of objc_get/setAssociatedObject.

**Pull request merged**